presto-matching/src/test/java/com/facebook/presto/matching/TestMatcher.java

```diff
@@ -70,11 +70,13 @@ public void propertyMatchers()
         Property<String, Integer> length = Property.property("length", String::length);
         String string = "a";
 
+        assertMatch(aString.with(length.equalTo(1)), string);
```

Comment:
Several reasons:
1. equals is a common case of a filter pattern one would expect to be available
2. it makes the patterns more readable
3. it allows for better toString at no cost
4. it secures optimization potential
5. it opens the way to building objects from patterns (remember the Builder and Generator interpreters from our recent discussion?)

for these reasons, I strongly believe equalTo patterns should stay.